"use strict";(self.webpackChunkeg3_docs=self.webpackChunkeg3_docs||[]).push([[931],{1106:(e,n,a)=>{a.r(n),a.d(n,{assets:()=>s,contentTitle:()=>o,default:()=>p,frontMatter:()=>c,metadata:()=>l,toc:()=>i});const l=JSON.parse('{"id":"datahub/example_datahub","title":"Example data hub","description":"Pasted below is an example data hub for mouse mm10:","source":"@site/docs/datahub/example_datahub.md","sourceDirName":"datahub","slug":"/datahub/example_datahub","permalink":"/datahub/example_datahub","draft":false,"unlisted":false,"editUrl":"https://github.com/twlab/eg3/tree/development/docs/docs/docs/datahub/example_datahub.md","tags":[],"version":"current","sidebarPosition":2,"frontMatter":{"sidebar_position":2},"sidebar":"tutorialSidebar","previous":{"title":"Datahub syntax","permalink":"/datahub/"},"next":{"title":"Track properties","permalink":"/datahub/track_properties"}}');var t=a(2540),r=a(3023);const c={sidebar_position:2},o="Example data hub",s={},i=[{value:"Example bigWig track",id:"example-bigwig-track",level:2},{value:"Example dynseq track",id:"example-dynseq-track",level:2},{value:"Example methylC track",id:"example-methylc-track",level:2},{value:"Example categorical track",id:"example-categorical-track",level:2},{value:"Example longrange track",id:"example-longrange-track",level:2},{value:"Example bigInteract track",id:"example-biginteract-track",level:2},{value:"Example repeatmasker track",id:"example-repeatmasker-track",level:2},{value:"Example geneAnnotation track",id:"example-geneannotation-track",level:2},{value:"Example bigbed track",id:"example-bigbed-track",level:2},{value:"Example bed track",id:"example-bed-track",level:2},{value:"Example refbed track",id:"example-refbed-track",level:2},{value:"Example HiC track",id:"example-hic-track",level:2},{value:"Example cool track",id:"example-cool-track",level:2},{value:"Example genomealign track",id:"example-genomealign-track",level:2},{value:"Example qBED track",id:"example-qbed-track",level:2},{value:"Example matplot track",id:"example-matplot-track",level:2},{value:"Example g3d track",id:"example-g3d-track",level:2},{value:"Example Ruler track",id:"example-ruler-track",level:2}];function d(e){const n={a:"a",admonition:"admonition",code:"code",h1:"h1",h2:"h2",header:"header",p:"p",pre:"pre",...(0,r.R)(),...e.components};return(0,t.jsxs)(t.Fragment,{children:[(0,t.jsx)(n.header,{children:(0,t.jsx)(n.h1,{id:"example-data-hub",children:"Example data hub"})}),"\n",(0,t.jsx)(n.p,{children:"Pasted below is an example data hub for mouse mm10:"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'[\n    {\n    "type": "bigwig",\n    "url": "https://vizhub.wustl.edu/public/tmp/TW463_20-5-bonemarrow_MeDIP.bigWig",\n    "name": "MeDIP",\n    "options": {\n        "color": "red",\n        "backgroundColor":"#FFE7AB"\n        },\n    "metadata": {\n        "sample": "bone",\n        "assay": "MeDIP"\n        }\n    },\n    {\n    "type": "bigwig",\n    "url": "https://vizhub.wustl.edu/public/tmp/TW551_20-5-bonemarrow_MRE.CpG.bigWig",\n    "name": "MRE",\n    "options": {\n        "color": "blue",\n        "backgroundColor":"#C0E3CC"\n        },\n    "metadata": {\n        "sample": "bone",\n        "assay": "MRE"\n        }\n    }\n]\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-bigwig-track",children:"Example bigWig track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "bigwig",\n    "name": "example bigwig",\n    "url": "https://vizhub.wustl.edu/hubSample/hg19/GSM429321.bigWig",\n    "options": {\n        "color": "blue"\n    }\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-dynseq-track",children:"Example dynseq track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "dynseq",\n    "name": "example dynseq",\n    "url": "https://target.wustl.edu/dli/tmp/deeplift.example.bw",\n    "options": {\n        "color": "blue",\n        "height: 100\n    }\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-methylc-track",children:"Example methylC track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n  "type": "methylc",\n  "name": "H1",\n  "url": "https://vizhub.wustl.edu/public/hg19/methylc2/h1.liftedtohg19.gz",\n  "options": {\n    "label": "Methylation",\n    "colorsForContext": {\n      "CG": { "color": "#648bd8", "background": "#d9d9d9" },\n      "CHG": { "color": "#ff944d", "background": "#ffe0cc" },\n      "CHH": { "color": "#ff00ff", "background": "#ffe5ff" }\n    },\n    "depthColor": "#01E9FE"\n  },\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-categorical-track",children:"Example categorical track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n  "type": "categorical",\n  "name": "ChromHMM",\n  "url": "https://egg.wustl.edu/d/hg19/E017_15_coreMarks_dense.gz",\n  "options": {\n      "category": {\n          "1": {"name": "Active TSS", "color": "#ff0000"},\n          "2": {"name": "Flanking Active TSS", "color": "#ff4500"},\n          "3": {"name": "Transcr at gene 5\' and 3\'", "color": "#32cd32"},\n          "4": {"name": "Strong transcription", "color": "#008000"},\n          "5": {"name": "Weak transcription", "color": "#006400"},\n          "6": {"name": "Genic enhancers", "color": "#c2e105"},\n          "7": {"name": "Enhancers", "color": "#ffff00"},\n          "8": {"name": "ZNF genes & repeats", "color": "#66cdaa"},\n          "9": {"name": "Heterochromatin", "color": "#8a91d0"},\n          "10": {"name": "Bivalent/Poised TSS", "color": "#cd5c5c"},\n          "11": {"name": "Flanking Bivalent TSS/Enh", "color": "#e9967a"},\n          "12": {"name": "Bivalent Enhancer", "color": "#bdb76b"},\n          "13": {"name": "Repressed PolyComb", "color": "#808080"},\n          "14": {"name": "Weak Repressed PolyComb", "color": "#c0c0c0"},\n          "15": {"name": "Quiescent/Low", "color": "#ffffff"}\n      }\n  }\n}\n'})}),"\n",(0,t.jsxs)(n.p,{children:["Supported options: ",(0,t.jsx)(n.a,{href:"#backgroundcolor",children:"backgroundColor"}),", ",(0,t.jsx)(n.a,{href:"#color",children:"color"}),", ",(0,t.jsx)(n.a,{href:"#color2",children:"color2"}),", ",(0,t.jsx)(n.a,{href:"#yscale",children:"yScale"}),", ",(0,t.jsx)(n.a,{href:"#ymax",children:"yMax"}),", and ",(0,t.jsx)(n.a,{href:"#ymin",children:"yMin"}),"."]}),"\n",(0,t.jsx)(n.h2,{id:"example-longrange-track",children:"Example longrange track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "longrange",\n    "name": "ES-E14 ChIA-PET",\n    "url": "https://egg.wustl.edu/d/mm9/GSE28247_st3c.gz"\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-biginteract-track",children:"Example bigInteract track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "biginteract",\n    "name": "test bigInteract",\n    "url": "https://epgg-test.wustl.edu/dli/long-range-test/interactExample3.inter.bb"\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-repeatmasker-track",children:"Example repeatmasker track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "repeatmasker",\n    "name": "RepeatMasker",\n    "url": "https://vizhub.wustl.edu/public/mm10/rmsk16.bb"\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-geneannotation-track",children:"Example geneAnnotation track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "geneAnnotation",\n    "name": "refGene",\n    "genome": "mm10"\n}\n'})}),"\n",(0,t.jsx)(n.admonition,{title:"note",type:"info",children:(0,t.jsxs)(n.p,{children:["Please specify the ",(0,t.jsx)(n.code,{children:"genome"})," attibute for gene annotation tracks."]})}),"\n",(0,t.jsx)(n.h2,{id:"example-bigbed-track",children:"Example bigbed track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "bigbed",\n    "name": "test bigbed",\n    "url": "https://vizhub.wustl.edu/hubSample/hg19/bigBed1"\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-bed-track",children:"Example bed track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "bed",\n    "name": "mm10 bed",\n    "url": "https://epgg-test.wustl.edu/d/mm10/mm10_cpgIslands.bed.gz"\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-refbed-track",children:"Example refbed track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "refbed",\n    "name": "mm10 gencode basic",\n    "url": "https://vizhub.wustl.edu/public/tmp/gencodeM18_load_basic_Gene.bed.gz",\n    "options": {\n            "categoryColors": {\n                "coding": "rgb(101,1,168)",\n                "nonCoding": "rgb(1,193,75)",\n                "pseudo": "rgb(230,0,172)",\n                "problem": "rgb(224,2,2)",\n                "other":"rgb(128,128,128)"\n            }\n        }\n}\n'})}),"\n",(0,t.jsx)(n.admonition,{title:"note",type:"info",children:(0,t.jsxs)(n.p,{children:[(0,t.jsx)(n.code,{children:"categoryColors"})," designates colors for the gene type as indicated in the 9th column. The default scheme is shown above for the five classes created by the ",(0,t.jsx)(n.code,{children:"Converting_Gencode_or_Ensembl_GTF_to_refBed.bash"})," script, but any number of categories can be defined."]})}),"\n",(0,t.jsx)(n.h2,{id:"example-hic-track",children:"Example HiC track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "hic",\n    "name": "test hic",\n    "url": "https://epgg-test.wustl.edu/dli/long-range-test/test.hic",\n    "options": {\n        "displayMode": "arc"\n    }\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-cool-track",children:"Example cool track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "cool",\n    "name": "Aiden et al. (2009) GM06900 HINDIII 1kb",\n    "url": "Hyc3TZevQVm3FcTAZShLQg",\n    "options": {\n        "displayMode": "arc"\n    }\n}\n'})}),"\n",(0,t.jsx)(n.admonition,{title:"note",type:"info",children:(0,t.jsxs)(n.p,{children:["please note we are using the uuid ",(0,t.jsx)(n.code,{children:"Hyc3TZevQVm3FcTAZShLQg"})," here from ",(0,t.jsx)(n.a,{href:"http://higlass.io/api/v1/tilesets",children:"higlass API server"})," instead of a file URL to represent a cool track."]})}),"\n",(0,t.jsx)(n.h2,{id:"example-genomealign-track",children:"Example genomealign track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "name": "hg19 to mm10 alignment",\n    "type": "genomealign",\n    "metadata": {\n        "genome": "mm10"\n    }\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-qbed-track",children:"Example qBED track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type":"qbed",\n    "url":"https://htcf.wustl.edu/files/RdNgrGeQ/HCT116-PBase.qbed.gz",\n    "name":"piggyBac insertions",\n    "showOnHubLoad":"true",\n    "options":{\n        "color":"#D12134",\n        "height":100,\n        "logScale":"log10",\n        "show":"sample",\n        "sampleSize":1000,\n        "markerSize":5,\n        "opacity":[50],\n      },\n}\n'})}),"\n",(0,t.jsx)(n.admonition,{title:"note",type:"info",children:(0,t.jsxs)(n.p,{children:["Default qBED track options are ",(0,t.jsx)(n.code,{children:'"logScale":"none"'}),", ",(0,t.jsx)(n.code,{children:'"show":"all"'}),", ",(0,t.jsx)(n.code,{children:'"markersize":3'}),", and ",(0,t.jsx)(n.code,{children:'"opacity":[100]'}),". Log-scaling can be set by specifying ",(0,t.jsx)(n.code,{children:'"logScale":"log10"'}),". To change opacity, pass a single value in brackets, as in the above example. qBED tracks will, by default, plot all entries in view. For information-dense regions, this can lead to excessive memory consumption. To plot a random subsample instead, specify ",(0,t.jsx)(n.code,{children:'"show":"sample"'})," and pass the number of points to visualize to ",(0,t.jsx)(n.code,{children:'"sampleSize"'}),", e.g. ",(0,t.jsx)(n.code,{children:'"sampleSize":1000'})]})}),"\n",(0,t.jsx)(n.h2,{id:"example-matplot-track",children:"Example matplot track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "matplot",\n    "name": "matplot wrap",\n    "tracks": [\n        {\n        "type": "bigwig",\n        "url": "https://vizhub.wustl.edu/public/tmp/TW463_20-5-bonemarrow_MeDIP.bigWig",\n        "name": "MeDIP",\n        "options": {\n            "color": "red",\n            "backgroundColor":"#FFE7AB"\n            },\n        "metadata": {\n            "sample": "bone",\n            "assay": "MeDIP"\n            }\n        },\n        {\n        "type": "bigwig",\n        "url": "https://vizhub.wustl.edu/public/tmp/TW551_20-5-bonemarrow_MRE.CpG.bigWig",\n        "name": "MRE",\n        "options": {\n            "color": "blue",\n            "backgroundColor":"#C0E3CC"\n            },\n        "metadata": {\n            "sample": "bone",\n            "assay": "MRE"\n            }\n        }\n    ]\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-g3d-track",children:"Example g3d track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "g3d",\n    "url": "https://wangftp.wustl.edu/~dli/tmp/test.g3d",\n    "name": "example 3d track",\n    "showOnHubLoad": true\n}\n'})}),"\n",(0,t.jsx)(n.h2,{id:"example-ruler-track",children:"Example Ruler track"}),"\n",(0,t.jsx)(n.pre,{children:(0,t.jsx)(n.code,{className:"language-json",children:'{\n    "type": "ruler",\n    "name": "Ruler"\n}\n'})})]})}function p(e={}){const{wrapper:n}={...(0,r.R)(),...e.components};return n?(0,t.jsx)(n,{...e,children:(0,t.jsx)(d,{...e})}):d(e)}},3023:(e,n,a)=>{a.d(n,{R:()=>c,x:()=>o});var l=a(3696);const t={},r=l.createContext(t);function c(e){const n=l.useContext(r);return l.useMemo((function(){return"function"==typeof e?e(n):{...n,...e}}),[n,e])}function o(e){let n;return n=e.disableParentContext?"function"==typeof e.components?e.components(t):e.components||t:c(e.components),l.createElement(r.Provider,{value:n},e.children)}}}]);